src/rp2_common/wifi_settings_connect/doc/SETTINGS_FILE.md

```diff
@@ -0,0 +1,160 @@
+# Creating and updating a WiFi settings file
+
+wifi\_settings\_connect stores WiFi hotspot names and passwords
+in a Flash sector that isn't normally used by programs. This
+is called the "WiFi settings file". It is similar to a file on a disk,
+except that it is always at the same location, and the size is
+limited to 4096 bytes.
+
+The file can be updated over USB by using [picotool](https://github.com/raspberrypi/picotool).
+It is a text file which can be edited with any text editor.
+Here is an example of typical contents:
+```
+    ssid1=MyHomeWiFi
+    pass1=mypassword1234
+    ssid2=MyPhoneHotspot
+    pass2=secretpassword
+    country=GB
+```
+wifi\_settings\_connect will automatically scan for hotspots and connect to
+hotspots matching the SSID names and passwords in the file.
+
+- On the [pico-wifi-settings home page](https://github.com/jwhitham/pico-wifi-settings)
+  you can also find a setup app which runs on your Pico and automates much of the
+  setup process. The pico-wifi-settings library is a superset of
+  wifi\_settings\_connect. It includes a remote update feature that allows
+  a new wifi-settings file to be installed via WiFi.
+
+# Creating the file on a computer
+
+Use any text editor to create a text file similar to the example above.
+
+Each line in the file should contain a key and a value, separated by `=`,
+with no spaces around `=`, or at the beginning or end of each line.
+
+The file must have at least `ssid1` or `bssid1`, otherwise there will
+be no connection attempts, and wifi\_settings\_connect will stay in the
+STORAGE\_EMPTY\_ERROR state.
+
+You can also use the following:
+
+ - `ssid<N>` - SSID name for hotspot N (a number from 1 to 100)
+ - `pass<N>` - Password for hotspot N
+ - `country` - Your two-letter country code from [ISO-3166-1](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes)
+ - `bssid<N>` - The BSSID ID for hotspot N
+ - `name` - The hostname of the Pico (sent to DHCP servers)
+
+# Copying the WiFi settings file by USB
+
+You can use [picotool](https://github.com/raspberrypi/picotool) to copy the
+file from your PC to your Pico via USB.
+
+picotool is part of the Pico SDK. 
+You need to build picotool with USB support for your OS (or download a pre-built copy).
```

Comment:
Might be worth linking to https://github.com/raspberrypi/pico-sdk-tools/releases here?